xsync/xsync/src/tasks/rust_toolchain_toml.rs

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use crate::Cmd;
+use anyhow::Context;
+use clap::Parser;
+use clap::Subcommand;
+
+#[derive(Subcommand)]
+pub enum Command {
+    /// Use base repo's `rust-toolchain.toml` to regenerate overlay's `rust-toolchain.toml`
+    Regen,
+}
+
+#[derive(Parser)]
+#[clap(
+    about = "Tools to keep rust-toolchain.toml files in-sync",
+    disable_help_subcommand = true
+)]
+pub struct RustToolchainToml {
+    #[clap(subcommand)]
+    pub cmd: Command,
+}
+
+impl Cmd for RustToolchainToml {
+    fn run(self, ctx: crate::CmdCtx) -> anyhow::Result<()> {
+        let Command::Regen = self.cmd;
+
+        let out = std::path::absolute(ctx.overlay_workspace.join("rust-toolchain.toml"))?;
+        let base_toolchain_toml =
+            fs_err::read_to_string(ctx.base_workspace.join("rust-toolchain.toml"));
+
+        match base_toolchain_toml {
+            Ok(base_toolchain_toml) => {
+                log::info!(
+                    "base rust-toolchain.toml found, regenerating overlay rust-toolchain.toml",
+                );
+                let base_toolchain_toml: schema::RustToolchainToml =
+                    toml_edit::de::from_str(&base_toolchain_toml)?;
+                let generated_toolchain_toml = format!(
+                    "{}{}",
+                    super::GENERATED_HEADER.trim_start(),
+                    toml_edit::ser::to_string_pretty(&base_toolchain_toml)?
+                );
+                log::debug!("{generated_toolchain_toml}");
+                fs_err::write(out, generated_toolchain_toml.as_bytes())?;
+            }
+            Err(e) if e.kind() == std::io::ErrorKind::NotFound => {
+                log::info!("base rust-toolchain.toml not found, removing overlay rust-toolchain.toml if present");
```

Comment:
is there any particular reason you chose this policy? it seems plausible that an overlay repo may want to maintain its own toolchain file, even on `main` branches, in cases where an internal toolchain should be used.

we don't actually do that in openvmm within microsoft today (we punt that toolchain selection logic to our CI), but it might make sense to rethink that policy choice in the future.

Reply:
The only reason was that we don't do it today. If the policy changes we can change the code.

Reply:
unless you feel strongly here, I'd rather keep this permissive and maintain whatever overlay policy is in-place when no base policy is in place

Reply:
I think I do, given that a rust-toolchain.toml file appearing in main is definitely a mistake today. I think I feel strongly enough that things should match between repos always, at least with today's policies.

Reply:
Ok. can you add a brief code comment summarizing this discussion in the code, to make it clear that this is a policy choice, and something we may want to tweak in the future?

Reply:
As we explore this space I imagine we could add config flags to this command to control this behavior and other aspects, but I don't think there's a point in speculating about what we may do in the future here until we actually have a reason to do it.